api_tests/users/views/test_user_institutions_relationship.py

```diff
@@ -0,0 +1,170 @@
+        res = self.app.get(
+            self.url, auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 200)
+
+        assert_in(self.user.institutions_self_url, res.json['links']['self'])
+        assert_in(self.user.institutions_related_url, res.json['links']['html'])
+
+        ids = [val['id'] for val in res.json['data']]
+        assert_in(self.institution1._id, ids)
+        assert_in(self.institution2._id, ids)
+
+    def test_get_institutions_relationship_while_logged_out(self):
+        res = self.app.get(
+            self.url
+        )
+        ids = [val['id'] for val in res.json['data']]
+        assert_in(self.institution1._id, ids)
+        assert_in(self.institution2._id, ids)
+
+    def test_post_with_auth(self):
+        res = self.app.post_json_api(
+            self.url, {},
+            auth=self.user.auth,
+            expect_errors=True
+        )
+
+        assert_equal(res.status_code, 405)
+
+    def test_put_with_auth(self):
+        res = self.app.put_json_api(
+            self.url, {},
+            auth=self.user.auth,
+            expect_errors=True
+        )
+
+        assert_equal(res.status_code, 405)
+
+    def test_post_without_auth(self):
+        res = self.app.post_json_api(
+            self.url, {}, expect_errors=True
+        )
+
+        assert_equal(res.status_code, 401)
+
+    def test_put_without_auth(self):
+        res = self.app.put_json_api(
+            self.url, {}, expect_errors=True
+        )
+
+        assert_equal(res.status_code, 401)
+
+    def test_delete_no_auth(self):
+        res = self.app.delete_json_api(
+            self.url,
+            {'data': [
+                {'type': 'institutions', 'id': self.institution1._id}
+            ]},
+            expect_errors=True
+        )
+
+        assert_equal(res.status_code, 401)
+
+    def test_delete_wrong_auth(self):
+        res = self.app.delete_json_api(
+            self.url,
+            {'data': [
+                {'type': 'institutions', 'id': self.institution1._id}
+            ]},
+            auth=self.user2.auth, expect_errors=True
+        )
+
+        assert_equal(res.status_code, 403)
+
+    def test_delete_one(self):
+        res = self.app.delete_json_api(
+            self.url,
+            {'data': [
+                {'type': 'institutions', 'id': self.institution1._id}
+            ]},
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 204)
+
+        self.user.reload()
+
+        ids = [inst._id for inst in self.user.affiliated_institutions]
+        assert_not_in(self.institution1._id, ids)
+        assert_in(self.institution2._id, ids)
+
+    def test_type_mistyped(self):
+        res = self.app.delete_json_api(
+            self.url,
+            {'data': [
+                {'type': 'wow', 'id': self.institution1._id}
+            ]},
+            auth=self.user.auth, expect_errors=True
+        )
+
+        assert_equal(res.status_code, 409)
```

Comment:
this is failing. Delete does not go through ` to_internal_value ` method in TypeField